lisa/sut_orchestrator/baremetal/cluster/idrac.py

```diff
@@ -0,0 +1,129 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import time
+import xml.etree.ElementTree as ETree
+from typing import Any, Type
+
+import redfish  # type: ignore
+
+from lisa import schema
+from lisa.environment import Environment
+from lisa.util import LisaException
+from lisa.util.logger import get_logger
+from lisa.util.perf_timer import create_timer
+
+from ..schema import ClusterSchema, IdracSchema
+from .cluster import Cluster
+
+
+class Idrac(Cluster):
+    def __init__(self, runbook: ClusterSchema) -> None:
+        super().__init__(runbook)
+        self.idrac_runbook: IdracSchema = self.runbook
+        self._log = get_logger("idrac", self.__class__.__name__)
+
+    @classmethod
+    def type_name(cls) -> str:
+        return "idrac"
+
+    @classmethod
+    def type_schema(cls) -> Type[schema.TypedSchema]:
+        return IdracSchema
+
+    def deploy(self, environment: Environment) -> Any:
+        self.redfish_instance = redfish.redfish_client(
+            base_url="https://" + self.idrac_runbook.address,
+            username=self.idrac_runbook.username,
+            password=self.idrac_runbook.password,
+        )
+        self._login()
+        self._change_boot_order_once("VCD-DVD")
+        for client in self.idrac_runbook.client:
+            assert client.iso_http_url, "iso_http_url is required for idrac client"
+            self._insert_virtual_media(client.iso_http_url)
+            self._reboot()
```

Comment:
Few comments:
1. I think it's better to change the boot order after virtual media insertion; just in case virtual media insertion fails, we don't want the server to have its boot order changed.
2. calling _eject_virtual_media() right before _insert_virtual_media() will also be good, as it will remove any stale inserted media.
3. After _reboot(), it's good to do sleep(<few seconds>) as _reboot does graceful shutdown and it might take sometime.
4.  I dont know if _exit() is implicitly called. Otherwise we should call _logout()

Reply:
For 1, I meant to do `self._change_boot_order_once("VCD-DVD")` after `_insert_virtual_media`. This is because we should be changing boot order only if the virtual media insertion is successful.

For 3 - Yes, we dont need to care about existing system, but server might not turn off immediately, I think it does graceful shutdown, and I've noticed cases where it takes some time for it to turn off. 
Since there is `ready_checker (ssh)` immediately after `deploy`, spur/paramiko might connect to the older existing system and LISA would think deploying is done. I have noticed this behavior when I was testing this PR for our usecases. Hence the suggestion to `sleep()` so that this is avoided. Another we could do here instead of sleep() is, check for the `reboot` job status through iDRAC api. 
I'm fine with using any of the above 2 methods.

Reply:
@LiliDeng Can we change the process to shutdown or turn off firstly, when it's done, then start it up? It should be more reliable.

Reply:
@LiliDeng Looks like point 1 is still not taken care.

For 3, I see you are doing  `self.stop()` and ` self.start()` . I think stop  is asynchronous operation. So should we do `_wait_for_completion` inside `stop` ?

Reply:
in _reset, it has _wait_for_completion, stop/start/reboot invoke _reset

Reply:
ok thanks.